processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -173,6 +176,14 @@ public Type getComponentType() {
         return componentType;
     }
 
+    public Type getBuilderType() {
+        return builderType;
+    }
+
+    public Type getMappingType() {
```

Comment:
Will add javadoc. A mapping Type is that type that should be used to find getters / setters and used in the generated code. I don't think we can use a util method as we are calling this from Freemarker and it would make it more difficult 😄

Reply:
ok.. the reason I mention this is because i'm always confused about the 'Result'- and 'Return'Type. So naming is important. What happens if we add another construct (next to builders) in the future?.. The problem I see is that `Mapping` is a bit overloaded. What about something like `getEffectiveType`? Anyway.. javadoc would help a lot as well.